### ISSUE TITLE
Proportional fonts are used for table output leading to column mis-alignment

### ISSUE THREAD

**Issue by sam-s** [millejoh/emacs-ipython-notebook#762]
## Problem description

pandas DataFrames are shown as tables:

![image](https://user-images.githubusercontent.com/1596894/102259107-b0827900-3edc-11eb-8899-8dd2317dad37.png)

as you can see, `+` and `|` are not aligned.

## Steps to reproduce the problem

```
import pandas as pd
df = pd.DataFrame({("a","x"):[1,2,3],("a","y"):[3,4,5],("b","x"):[5,6,7],"i":[9,8,7]}).set_index("i")
df.columns=pd.MultiIndex.from_tuples(df.columns)
df
```

## System info:

```cl
("EIN system info"
 :emacs-version
 "GNU Emacs 28.0.50 (build 1, x86_64-apple-darwin19.6.0, NS appkit-1894.60 Version 10.15.7 (Build 19H15))
 of 2020-12-14"
 :emacs-bzr-version "e8a358c3be90949645a1038cfd43553794c49441"
 :window-system ns
 :emacs-variant nil
 :os
 (:uname
  "Darwin BZ-C02XR5CGJG5L 19.6.0 Darwin Kernel Version 19.6.0: Thu Oct 29 22:56:45 PDT 2020; root:xnu-6153.141.2.2~1/RELEASE_X86_64 x86_64
"
  :lsb-release nil)
 :notebook
 "Name: notebook
Version: 6.1.5
Summary: A web-based notebook environment for interactive computing
Home-page: http://jupyter.org
Author: Jupyter Development Team
Author-email: jupyter@googlegroups.com
License: BSD
Location: /Users/sam.steingold/.virtualenvs/brazeds/lib/python3.9/site-packages
Requires: jupyter-client, nbconvert, ipykernel, jinja2, ipython-genutils, argon2-cffi, jupyter-core, pyzmq, prometheus-client, nbformat, tornado, terminado, Send2Trash, traitlets
Required-by: widgetsnbextension, jupyter, jupyter-nbextensions-configurator, jupyter-latex-envs, jupyter-contrib-nbextensions, jupyter-contrib-core
"
 :ipython "7.19.0
"
 :image-types
 (imagemagick png gif tiff jpeg xpm xbm pbm)
 :image-types-available
 (imagemagick png gif tiff jpeg xpm xbm pbm)
 :request
 (:backend curl)
 :ein
 (:version "3913fda-dirty"
           :source-dir "/Users/sam.steingold/src/sitelisp/emacs-ipython-notebook/lisp/")
 :lib
 ((:name "websocket"
         :path "~/.config/emacs/elpa/websocket-20201122.246/websocket.elc"
         :featurep t
         :version-var websocket-version
         :version "1.12")
  (:name "anaphora"
         :path "~/.config/emacs/elpa/anaphora-20180618.2200/anaphora.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "request"
         :path "~/.config/emacs/elpa/request-20201026.2324/request.elc"
         :featurep t
         :version-var request-version
         :version "0.3.0")
  (:name "deferred"
         :path "~/.config/emacs/elpa/deferred-20170901.1330/deferred.elc"
         :featurep t
         :version-var deferred:version
         :version "0.5.0")
  (:name "polymode"
         :path "~/.config/emacs/elpa/polymode-20200606.1106/polymode.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "dash"
         :path "~/.config/emacs/elpa/dash-20200803.1520/dash.elc"
         :featurep t
         :version-var nil
         :version nil)
  (:name "with-editor"
         :path "~/.config/emacs/elpa/with-editor-20201030.1232/with-editor.elc"
         :featurep nil
         :version-var nil
         :version nil)))
```

**Comment by dickmao** [comment]
Yes, I knew commit a019ee3 was a bad idea (it chose html outputs over plaintext outputs in trying to please #756 ).

Thanks for reporting.
